src/serac/numerics/functional/domain.hpp

```diff
@@ -156,4 +226,34 @@ inline auto by_attr(int value)
   return [value](std::vector<tensor<double, dim> >, int attr) { return attr == value; };
 }
 
+/**
+ * @brief count the number of elements of each geometry in a domain
+ * @param domain the domain to count
+ */
+inline std::array<uint32_t, mfem::Geometry::NUM_GEOMETRIES> geometry_counts(const Domain& domain)
+{
+  std::array<uint32_t, mfem::Geometry::NUM_GEOMETRIES> counts{};
+
+  constexpr std::array<mfem::Geometry::Type, 5> geometries = {mfem::Geometry::SEGMENT, mfem::Geometry::TRIANGLE,
+                                                              mfem::Geometry::SQUARE, mfem::Geometry::TETRAHEDRON,
+                                                              mfem::Geometry::CUBE};
+  for (auto geom : geometries) {
+    counts[uint32_t(geom)] = uint32_t(domain.get(geom).size());
+  }
+  return counts;
+}
+
+/**
+ * @brief convenience function for computing the arithmetic mean of some list of vectors
+ */
+template <int dim>
+inline tensor<double, dim> average(std::vector<tensor<double, dim> >& positions)
```

Comment:
I think this has been reimplemented locally in a few places.  Maybe already removed duplication, or can be something for the future?

Reply:
this is the deduplication of those other functions-- I've removed all of them (that I found) and put the common implementation here